src/types/json.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#pragma once
+
+#include <jsoncons/json.hpp>
+#include <jsoncons/json_error.hpp>
+#include <jsoncons_ext/jsonpath/json_query.hpp>
+
+#include "jsoncons_ext/jsonpath/jsonpath_error.hpp"
+#include "status.h"
+
+struct JsonValue {
+  JsonValue() = default;
+  explicit JsonValue(jsoncons::basic_json<char> value) : value(std::move(value)) {}
+
+  static StatusOr<JsonValue> FromString(std::string_view str) {
+    jsoncons::json val;
+
+    try {
+      val = jsoncons::json::parse(str);
+    } catch (const jsoncons::ser_error &e) {
+      return {Status::NotOK, e.what()};
+    }
+
+    return JsonValue(std::move(val));
+  }
+
+  std::string Dump() const { return value.to_string(); }
+
+  Status Set(std::string_view path, JsonValue &&new_value) {
+    try {
+      jsoncons::jsonpath::json_replace(value, path, [&new_value](const std::string & /*path*/, jsoncons::json &origin) {
+        origin = std::move(new_value.value);
+      });
```

Comment:
I think since it follows *preorder traversal* so it will goes well when there are mutlple sub json objects matched in the original object.

But it will be wrong if there are mutlple sub json objects matched in the new_value, which will be fixed later.

Reply:
I don't think so, since `new_value.value` has been moved.

Reply:
Oh, good point. I will fix it.